### ISSUE TITLE
Using from a Windows Service

### ISSUE THREAD

**Issue by jnl002** [alphaleonis/AlphaVSS#46]
Does the AlphaVSS code work when running from within a Windows Service?

In my tests, it seems to crash on this line when I try to run it from within a .NET Service project.

IVssImplementation _vssImplementation = VssUtils.LoadImplementation();

It works perfectly if I run out from a normal Windows process.

Thanks in advance for any tips.

John.

**Comment by alphaleonis** [comment]
I don't see any reason why it wouldn't. What is the error message/exception you are encountering?

**Comment by jnl002** [comment]
I'm not sure what the error message is since I have not managed to successfully write the error out to a log. Would you like a sample of the project? If so, what would be the best way to upload the project here?

Thanks.

John.

**Comment by alphaleonis** [comment]
You could just do `File.WriteAllText("C:\\errors.txt", errorMessage)` as a poor mans logging, or attach a debugger and see what happens (in a service for example by putting `Debugger.Launch()` at the start of the code somewhere and catch the error that way to see what it is.  

Otherwise you can attach files to posts in issues on GitHub just by dragging your files here, but I'm not sure I have got the time to personally analyze it unfortunately.

**Comment by jnl002** [comment]
Thanks for your help, alphaleonis.

I managed to detect what was wrong. It was a novice error. I was accidently executing the main process from a folder that was missing the required DLL files. Once I fixed that, the calls to VSS worked just fine when running as a Windows service.

This may be another newbie question, but I would like to verify that these files can be freely distributed within my custom applications without any royalty requirements. Is that the case? I just wanted to ask in order to be on the safe side.

Thanks again.

John.
